doc/manual/src/command-ref/files/channels.md

```diff
@@ -1,26 +1,41 @@
 ## Channels
 
-A directory containing symlinks to Nix channels, managed by [`nix-channel`]:
-
-- `$XDG_STATE_HOME/nix/profiles/channels` for regular users
-- `$NIX_STATE_DIR/profiles/per-user/root/channels` for `root`
-
-[`nix-channel`] uses a [profile](@docroot@/command-ref/files/profiles.md) to store channels.
-This profile contains symlinks to the contents of those channels.
+Channels are a mechanism for obtaining Nix expressions over the web.
+
+The moving parts of channels are:
+- The official channels listed at <https://nixos.org/channels>
+- The user-specific list of [subscribed channels](#subscribed-channels)
+- The [downloaded channel contents](#channel-contents)
+- The [Nix expression search path](@docroot@/command-ref/conf-file.md#conf-nix-path)
+
+> **Note**
+>
+> The state of a subscribed channel is external to the Nix expressions relying on it.
+> This may limit reproducibility.
```

Comment:
We can confidently state that this is a problem for reproducibility and make it slightly more actionable.

```suggestion
> The state of a subscribed channel is external to the Nix expressions relying on it.
> This limits reproducibility, and may be a reason to use a lock file instead of channels.
```

I would like to link a resource that helps with deciding the kind of lock file, but I don't want to derail this review, so I'd prefer to settle for an incremental improvement here, as far as my input is concerned. I'd be happy to accept a suggestion.

Reply:
We'd be opening up quite the rabbit hole here. I generally agree with the direction, but my aspiration would be to point people to The Right Way of doing things, and that to my knowledge still doesn't exist. The current version says something that is at least not wrong: use fetchers explicitly.

Reply:
I guess a pin is not necessarily in a lock file. Fair enough.
We could confidently say that pinning and/or locking is The Right Way, but it might be a distraction.

Initially I thought nix.dev had a suitable article about this, but they seem to be about specific solutions, which is good for tutorials; was looking for a guide.

Let's not get ourselves stuck on this and fill this gap later.